gateway-ha/src/main/java/io/trino/gateway/ha/security/LbLdapParser.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.gateway.ha.security;
+
+import io.trino.gateway.ha.config.AuthorizationConfiguration;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class LbLdapParser
+{
+    private final AuthorizationConfiguration authorizationConfiguration;
+
+    public LbLdapParser(AuthorizationConfiguration authorizationConfiguration)
+    {
+        this.authorizationConfiguration = authorizationConfiguration;
+    }
+
+    /**
+     * The input is the ldap search from the user. It's in the format of
+     * String memberOf = """
+     * memberOf: ...
+     * memberOf: ...
+     * """
```

Comment:
Am not a ldap expert, but how are you sure that it is `memberOf: {...}` format?
I think this depends on how LDAP server returns with how we set ldapGroupMemberAttribute. 


`ldapGroupMemberAttribute: 'ou'` shows `ou: memberof={}`
![image](https://github.com/trinodb/trino-gateway/assets/25147023/ddc83b94-260e-4c8b-bb47-538e6e50c7c1)

`ldapGroupMemberAttribute: 'uid'` shows `uid: {}`
![image](https://github.com/trinodb/trino-gateway/assets/25147023/7c751315-6289-44dd-b958-e348cb091a74)

Reply:
right... there's some assumption made in this